[Lew Kai Ming] Duke Increments

KxxMxxx opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

Commit:
Merge branch 'gradle'

Commit:
Level 1

Commit:
Level 2 done

Commit:
Level 3 done

Commit:
Level 4 done

Commit:
Level 5 done

Commit:
Level 6 done

Commit:
added level 6 text ui testing files

Commit:
Cleaned up code

Commit:
Level 7 done

Commit:
Level 8 done

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
A-MoreOOP increment done

Commit:
Minor changes made to duke and storage

Commit:
increment A-JUnit done

Commit:
.gitignore file committed.

Commit:
added additional test class

Commit:
increment A-JavaDoc done

Commit:
branch-Level-9 done

Commit:
modify test and add additional test

Previously encountered errors with JUnit testing. Re-imported JUnit into
project and added additional test that works with Duke.

Commit:
tidy to align with coding standards

Commit:
tidy javadocs to align with coding standards

Commit:
rename package from Duke to duke

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
refactor: refactor TaskList

TaskList can find based on multiple keywords

chanjunren commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+/**
+ * A Parser class parses all the user input.
+ */
+public class Parser {
```
Hi bro, 
it might be better if you make use of a command class rather than using the parser to call the other classes :D But your code is very clean and easy to read

Commit:
refactor: fix exceptions and tidy code

Add DukeTaskException, DukeParserException and DukeListException. Minor
fix to code to align with coding standards.

Commit:
refactor: shift main logic to duke class

Commit:
initial level-10 commit

Commit:
Merge branch 'branch-Level-10'

xinmingzh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,67 @@
+        while(!isExit) {
+            String input = sc.nextLine();
+            if(input.equals("bye")) {
+                System.out.println(parser.parse(input));
```
Why not consolidate duplicate code?

KxxMxxx replied:
By that do you mean to abstract out the System.out.println()? Could you suggest how I can go about doing that? Thanks!

xinmingzh replied:
I am referring to line 47 and 50 can be simplified to one line above the if else clause.

KxxMxxx replied:
Oh ok I see what you meant, thanks!

xinmingzh commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,125 @@
+                        tasks.add(event);
+                    } else {
+                        tasks.add(event);
+                    }
```
Why not consolidate all the tasks.add(events) ie. call the method once after the switch?

KxxMxxx replied:
That's a good suggestion! I could use polymorphism and use a Task reference to refer to the object instances and add it after the switch.

xinmingzh reviewed (commented):
There seems to be some duplicate code which I prefer to consolidate. Is there any considerations when you did it that way? Otherwise, everything seems clean and organised.

KxxMxxx commented:
I used to do the printing in the UI class, however, to compromise for the addition of JavaFx, I had to return Strings instead. That's why I had to put the System.out.println() in the Duke class instead.

chanjunren reviewed (commented):
(Sorry I forgot to press submit review)

Commit:
Add command and improve code logic.

Code is more logical and OOP after addition of command class.

Commit:
Fix checkstyle errors of code

KxxMxxx committed:
Add assertions for A-Assertions (#4)

Commit:
Correct code quality issues

Simplified complicated expressions and shorten long methods.

KxxMxxx committed:
Add alias command to support friendlier syntax (#6)

Aliases to the basic commands can be added using the Alias command.

Commit:
add help command

user can type help to see a guide to use Duke

Commit:
add save and load function for aliases

Commit:
clean up JavaFx

Commit:
make JavaFx not resizable

Commit:
add file directory and file creation to storage

Commit:
create user guide

KxxMxxx committed:
Set theme jekyll-theme-minimal

Commit:
add line breaks to user guide

Commit:
Merge branch 'master' of https://github.com/KxxMxxx/duke

Commit:
fix formatting on user guide

Commit:
change main class in build.gradle to Launcher
